cishouseholds/weights/weights.py

```diff
@@ -398,17 +402,24 @@ def validate_design_weights_or_precal(
     - dweights consistent by cis area
     """
     # import pdb; pdb.set_trace()
-    window = Window.partitionBy(F.lit(1))
+    cis_window = Window.partitionBy(cis_area_column)
+    country_window = Window.partitionBy(country_column)
+
     columns = [col for col in df.columns if "weight" in col and list(df.select(col).dtypes[0])[1] == "double"]
     # check 1.1
     df = df.withColumn(
         "CHECK1s",
-        F.when(F.sum(swab_weight_column).over(window) == F.col(num_households_by_cis_column), 0).otherwise(1),
+        F.when(
+            F.round(F.sum(swab_weight_column).over(cis_window), 0) == F.col(num_households_by_cis_column), 0
```

Comment:
Methodology have said that this rounding shouldn't be necessary. Perhaps when the data have been corrected these will sum correctly
I noticed yesterday that Null country or CIS area is treated as it's own group, so I'd imagine that is causing issues while we have Null values in the geographies

Reply:
true, that may be skewing the count